src/nvim/api/buffer.c

```diff
@@ -179,6 +179,11 @@ void buffer_set_line_slice(Buffer buffer,
     return;
   }
 
+  if (!inbounds(buf, start)) {
```

Comment:
Same as above, how to make changes to the end of the buffer? For client-initiated "conversations", nvim doesn't block between requests, so one could potentially use an invalid line counts.